rfcs/0128-redeployable-clients.md

```diff
@@ -0,0 +1,93 @@
+# RFC 0128 - Provide docs/standards about client/worker/proxy interactions in redeployable taskcluster
+* Comments: [#128](https://api.github.com/repos/taskcluster/taskcluster-rfcs/issues/128)
+* Proposed by: @petemoore
+
+# 1. Summary
+
+Changes are required for clients of taskcluster services when taskcluster
+becomes redeployable.
+
+This RFC serves to define the standards by which taskcluster deployments will
+publish a manifest of their service definitions, how client generators will
+query and interpret this manifest and associated reference and schema documents
+in order to generate clients, what the architectural changes to generated
+clients will be, how users of those clients (workers, worker authentication
+proxies, command line tools, software libraries) will be adapted in order to
+use the new clients, and how generated clients and client generators will be
+built, released and deployed.
+
+This entails changes to the way workers will share deployment / authentication
+proxy configuration information with tasks, versioning of the API / Event
+reference schemas, versioning of the services definitions based on those API
+and Event reference schemas, and the possibility for clients to connect to
+different taskcluster deployments.
+
+## 2. Motivation
+
+This RFC is needed in order to support the redeployability project (multiple
+deployments of taskcluster rather than just a single global deployment under
+taskcluster.net domain).
+
+# 3. Details
+
+## 3.1. Publishing API schemas ("references")
+
+### 3.1.1 Current situation
+
+Taskcluster services currently provide either one or two of the following types
+of interface (API):
+
+* HTTP (mostly RESTful)
+* AMQP 0.9.1 (provided via mozilla pulse, a customised RabbitMQ deployment)
```

Comment:
These actually just work with rabbitmq directly in the redeployed cluster at least. No need for pulse!

Reply:
We look at Pulse as a particular set of [expectations for how to run a RabbitMQ cluster](https://wiki.mozilla.org/Auto-tools/Projects/Pulse).

Reply:
It might be easiest to just use the term "Events API" for this.  I don't think we're tied to a specific AMQP version, for example.